tests/e2e/internal/framework/kube.go

```diff
@@ -92,6 +92,12 @@ func (c *KubeClient) CheckPodCondition(ctx context.Context,
 		}
 	}
 
+	for _, c := range pod.Status.ContainerStatuses {
+		if c.State.Waiting != nil && c.State.Waiting.Reason == "CrashLoopBackOff" {
+			return false, fmt.Errorf("pod %s in namespace %s is in CrashLoopBackOff", pod.Name, pod.Namespace)
+		}
+	}
+
```

Comment:
Not clear why you also need to query containr status.
Also, this breaks the single responsibility rule now this function is doing two different things: 1. Checking pod status, 2. checking underlying container statuses.